src/nfa/nfa.cc

```diff
@@ -456,6 +456,10 @@ void Nfa::print_to_dot(std::ostream &output, const bool decode_ascii_chars, cons
         if (decode_ascii_chars) {
             return to_ascii(symbol);
         }
+        if (this->alphabet != nullptr) {
```

Comment:
This should be `else if`, as the decoding should take precedence, then the alphabet passed as a parameter, and then the member alphabet.